### ISSUE TITLE
node-require module removes .marko extension from filenames in compiled code.

### ISSUE THREAD

**Issue by lohfu** [marko-js/marko#262]
when using the `require('marko').install()` the compiled files seem to cut the .marko extension from all require.resolve() calls.

When i have compiled using `$ markoc master.marko` my output contains:
```
navigation_template = loadTemplate(require.resolve("./components/navigation/template.marko"));
```
but when i have installed the node-require module, and simply required the template with 
`require('./src/master.marko')` the same line is:
```
navigation_template = loadTemplate(require.resolve("./components/navigation/template"));
```

**Comment by patrick-steele-idem** [comment]
Thanks for reporting the problem. I'm looking into it.

**Comment by patrick-steele-idem** [comment]
@lohfu I was not able to reproduce the problem using a fresh `npm install`. I tried to recreate your setup and the template is compiling as expected:

```javascript
function create(__helpers) {
  var str = __helpers.s,
      empty = __helpers.e,
      notEmpty = __helpers.ne,
      escapeXml = __helpers.x,
      loadTemplate = __helpers.l,
      navigation_template = loadTemplate(require.resolve("./components/navigation/template.marko"));

  return function render(data, out) {
    navigation_template.render({}, out);
  };
}

(module.exports = require("marko").c(__filename)).c(create);
```

**Comment by lohfu** [comment]
Sweet. As soon as that work as expected i think i have a functioning pull
request for markoify.

**Comment by lohfu** [comment]
Ok, i will see if i can recreate it again.

**Comment by patrick-steele-idem** [comment]
Hey @lohfu, just wanted to check in and see if you were able to reproduce. Thanks.

**Comment by lohfu** [comment]
Hey Patrick,

Yes i was. I created a repository at https://github.com/lohfu/marko-test.
Simply running `node server/server.js` will create a master.marko.js with

```
navigation_template =
loadTemplate(require.resolve("./components/navigation/template")),
```

**Comment by patrick-steele-idem** [comment]
Thanks for creating the repo... I was able to reproduce the problem and am looking into it this morning. I'll update when I know more. Thanks again.

**Comment by patrick-steele-idem** [comment]
Hey @lohfu, I realized later that I was trying to reproduce the problem using the `<include>` tag. Once I saw that you were using a custom tag mapped to a template it was easy to reproduce and easy to fix. Technically, everything was working as expected, but it would cause problems if a transform was looking at the file extension for the `require`/`require.resolve` path and not the fully resolved path. 

Anyway, the problem was fixed. New version published: `marko@3.1.5`

Please let me know if you see any other problems. Thanks!
